Sources/SwiftBuildSupport/PackagePIFProjectBuilder+Modules.swift

```diff
@@ -762,18 +763,27 @@ extension PackagePIFProjectBuilder {
             }
         }
 
-        // Set the imparted settings, which are ones that clients (both direct and indirect ones) use.
-        var debugImpartedSettings = impartedSettings
-        debugImpartedSettings[.LD_RUNPATH_SEARCH_PATHS] =
-            ["$(BUILT_PRODUCTS_DIR)/PackageFrameworks"] +
-            (debugImpartedSettings[.LD_RUNPATH_SEARCH_PATHS] ?? ["$(inherited)"])
+        // Set the **imparted** settings, which are ones that clients (both direct and indirect ones) use.
+        // For instance, given targets A, B, C with the following dependency graph:
+        //
+        //   A (executable) -> B (dynamicLibrary) -> C (objectFile)
+        //
+        // An imparted build setting on C will propagate back to both B and A.
+        let additionalRunPaths = if productType == .framework {
+            ["$(BUILT_PRODUCTS_DIR)/PackageFrameworks"]
```

Comment:
This is changing the behavior a bit from previously since the settings are no longer being added to a separate `debugImpartedSettings` variable. Since `$(BUILT_PRODUCTS_DIR)/PackageFrameworks` is an absolute path to the build outputs, that should only be present in debug configurations, not in release configurations.

`@loader_path` on the other hand should be present in both places.

Reply:
Hmm... makes total sense now! 👍🏼